sdk/include/sdk/Node.h

```diff
@@ -48,15 +60,24 @@ class Node {
      */
     [[nodiscard]] std::string getPath() const;
 
+    [[nodiscard]] virtual Type getType() const { return Type::BRANCH; }
+
+    [[nodiscard]] virtual const DataPoint* getDataPoint(const std::string& path) const;
+
     Node(const Node&)            = delete;
     Node(Node&&)                 = delete;
     Node& operator=(const Node&) = delete;
     Node& operator=(Node&&)      = delete;
 
+protected:
+    void registerChild(const Node& childNode);
+
 private:
     // TODO: Use std::weak_ptr ?
-    Node*       m_parent;
-    std::string m_name;
+    Node* const       m_parent;
```

Comment:
Do we have any style guide on how long indentation we have before the variable/member name? Like why do we have multiple here but only one before `m_children`?